staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -78,6 +78,7 @@ var (
 	// We try to spread the load on apiserver by setting timeouts for
 	// watch requests - it is random in [minWatchTimeout, 2*minWatchTimeout].
 	minWatchTimeout = 5 * time.Minute
+	ErrStaleWatch   = apierrs.NewResourceExpired("The resourceVersion for the provided watch is too old.")
```

Comment:
https://github.com/kubernetes/kubernetes/blob/b3875556b0edf3b5eaea32c69678edcf4117d316/staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go#L467

consider reuse the error for the watch-cache?

Reply:
I fear we cannot access that here from client-go.